documentation/staging/content/userguide/managing-domains/model-in-image/common-mounts.md

```diff
@@ -0,0 +1,444 @@
+  sample-domain1-admin-server   1/1   Running   0     32s
+  sample-domain1-managed-server1   0/1   Pending   0     0s
+  sample-domain1-managed-server2   0/1   Pending   0     0s
+  sample-domain1-managed-server1   0/1   ContainerCreating   0     0s
+  sample-domain1-managed-server2   0/1   ContainerCreating   0     0s
+  sample-domain1-managed-server1   0/1   Running   0     2s
+  sample-domain1-managed-server2   0/1   Running   0     2s
+  sample-domain1-managed-server1   1/1   Running   0     43s
+  sample-domain1-managed-server2   1/1   Running   0     42s
+  ```
+  {{% /expand %}}
+
+Alternatively, you can run `/tmp/mii-sample/utils/wl-pod-wait.sh -p 3`. This is a utility script that provides useful information about a domain's pods and waits for them to reach a `ready` state, reach their target `restartVersion`, and reach their target `image` before exiting.
+
+  {{%expand "Click here to display the `wl-pod-wait.sh` usage." %}}
+  ```shell
+    $ ./wl-pod-wait.sh -?
+    ```
+    ```
+    Usage:
+
+      wl-pod-wait.sh [-n mynamespace] [-d mydomainuid] \
+         [-p expected_pod_count] \
+         [-t timeout_secs] \
+         [-q]
+
+      Exits non-zero if 'timeout_secs' is reached before 'pod_count' is reached.
+
+    Parameters:
+
+      -d <domain_uid> : Defaults to 'sample-domain1'.
+
+      -n <namespace>  : Defaults to 'sample-domain1-ns'.
+
+      -p 0            : Wait until there are no running WebLogic Server pods
+                        for a domain. The default.
+
+      -p <pod_count>  : Wait until all of the following are true
+                        for exactly 'pod_count' WebLogic Server pods
+                        in the domain:
+                        - ready
+                        - same 'weblogic.domainRestartVersion' label value as
+                          the domain resource's 'spec.restartVersion'
+                        - same 'weblogic.introspectVersion' label value as
+                          the domain resource's 'spec.introspectVersion'
+                        - same image as the the domain resource's image
+
+      -t <timeout>    : Timeout in seconds. Defaults to '1000'.
+
+      -q              : Quiet mode. Show only a count of wl pods that
+                        have reached the desired criteria.
+
+      -?              : This help.
+  ```
+  {{% /expand %}}
+
+  {{%expand "Click here to view sample output from `wl-pod-wait.sh`." %}}
+  ```
+  @@ [2020-04-30T13:50:42][seconds=0] Info: Waiting up to 1000 seconds for exactly '3' WebLogic Server pods to reach the following criteria:
+  @@ [2020-04-30T13:50:42][seconds=0] Info:   ready='true'
+  @@ [2020-04-30T13:50:42][seconds=0] Info:   image='model-in-image:WLS-v1'
+  @@ [2020-04-30T13:50:42][seconds=0] Info:   domainRestartVersion='1'
+  @@ [2020-04-30T13:50:42][seconds=0] Info:   namespace='sample-domain1-ns'
+  @@ [2020-04-30T13:50:42][seconds=0] Info:   domainUID='sample-domain1'
```

Comment:
This is incorrect. It's a cut-and-paste from the old documentation for the non-common-mount case.

There are several things going on here that look like they need some investigation:

1) The current version of wl-pod-wait.sh 'image' is expected to be the base image since the script obtains the image name from the resource, and the resource image is the base image in the common mount use case.   So the 'image='model-in-image:WLS-v1'' output from wl-pod-wait.sh does not look it was run with the common mount use case.

2) On a different, but related note, the wl-pod-wait.sh script should ideally be updated to detect when the common mount image loaded by each init container matches the common mount image specified under 'serverPod' stanza for each pod.  I don't know if this information is even readily available on each pod, and even if it is available, I don't know if it'd be too complex to update the script to do this.  This is a problem, because if the common mount image is changed without also changing the domainRestartVersion, then the wl-pod-wait script can incorrectly assume that everything is up to date when it really isn't...

Reply:
1 has been addressed, 2 is still pending (am looking into it)